### ISSUE TITLE
Ventricle alignment NKI data

### ISSUE THREAD

**Issue by amanbadhwar** [SIMEXP/niak#100]
Need to fine-tune ventricle alignment for the NKI data

Ventricle issue (example cases_they might or might not have other issues):
s0152384
s0159429
s0180093
s0105290
s0199620
s0146865 - the transformation was done on this one as an example
s0194956
s0113013
s0112828
s0198130
s0169007
s0186697
s0109819

**Comment by pbellec** [comment]
please post a picture.

**Comment by amanbadhwar** [comment]
Pictures
![0146865_f_ventricle_issue](https://cloud.githubusercontent.com/assets/11929551/11045757/6b3b28bc-86f7-11e5-8336-8924e7d152cf.jpg)
![0146865_f_ventricle_issue_2](https://cloud.githubusercontent.com/assets/11929551/11045772/8609d22e-86f7-11e5-9015-d3ec8a253799.jpg)

**Comment by pbellec** [comment]
So I have identified a combination of two issues:
1. the signal loss in ventral frontal cortex for BOLD images. That one's pretty obvious but I had underestimated the bias it could induce in registration results.
2. A more subtle one. Some subjects have a lot of csf between the brain and the skull. This is (correctly) masked out as non-brain tissue in the T1 image. But it does generate a lot of signal in the BOLD images, while typically only the brain has high signal. So in this case the T1 and BOLD brain masks do not match. 

Interestingly 1 and 2 can combine to generate a misregistration, as the CSF space often happens quite dorsally. The registration pushes the CSF inside the cortex to match the edges of the masks, and also compensate the signal loss ventrally. A perfect trap. Working on a solution, I think I have something that should work pretty accurately and robustly. Building in the issue100 branch.

**Comment by pbellec** [comment]
I have implemented a solution around two ideas (1) add the CSF that lies between the brain and the skull into the T1 mask for registration. (2) Eliminate voxels that typically show signal dropout in BOLD at 3T from the T1 mask for registration. Each of these mechanism can be switched on/off with dedicated flags, and the second one also has a couple parameters to adjust its behaviour. Will detail a doc and recommended tweaking if it proves to work well. I have preprocessed Cambridge with this trick, and checked the ten first subjects and registrations were close to perfect for all of them (with the exception of one which had a small T1 registration problem but this was not related to the BOLD-T1 registration). @amanbadhwar Can you please run 10-15 problematic subjects from NKI enhanced with the proposed fix? You don't need to change anything to your script. Just use the following quarantine on guillimin `quarantaine/niak-issue100`

**Comment by pbellec** [comment]
alright so I implemented the idea, checked it fixed a highly problematic case, then checked it worked well (or better) than the old version on 20 subjects from the Cambridge sample, then @amanbadhwar checked it fixed problematic subjects from the NKI enhanced sample. So it seems to be an improvement, I just merged it to master with the help of @poquirion who generated a new target. Will require more testing but very promising. Also, we'll need some doc (it's possible to remove parts or all of the fix from the pipeline, or tweak the ammount of ventral signal loss). I'll create a separate issue for that.
